storage/src/desktop/storage_reference_desktop.cc

```diff
@@ -252,6 +255,19 @@ void StorageReferenceInternal::PrepareRequest(rest::Request* request,
   // Add the Firebase library version to the request.
   request->add_header("X-Firebase-Storage-Version",
                       storage_->user_agent().c_str());
+
+  // Use the function registry to get the App Check token.
+  Future<::firebase::app_check::AppCheckToken> app_check_future;
+  bool succeeded = storage_->app()->function_registry()->CallFunction(
+      ::firebase::internal::FnAppCheckGetTokenAsync, storage_->app(), nullptr,
+      &app_check_future);
+  if (succeeded && app_check_future.status() != kFutureStatusInvalid) {
+    const ::firebase::app_check::AppCheckToken* token =
+        app_check_future.Await(30000);
```

Comment:
Probably a good idea to make this 30000 a constant with a name that clarifies that this is a number of milliseconds
30s seems like potentially a pretty long time to wait. Do we wait for a similar duration for other operations like getting auth credentials?

Reply:
So, Auth tends to not wait for the token, and just sorta assumes it will be there, which I didn't want to do. I did set it to 10 seconds instead, and moved it out along with a comment.